### PR TITLE
Update crossbell extension

### PR DESCRIPTION
**Pull request by hyoban**
## Description

This PR update all dependencies, and add a new command `upload-file`.
It can upload files from your system clipboard or local file to IPFS.

## Screencast

From clipboard

https://github.com/raycast/extensions/assets/38493346/8df55c39-3ab3-4a52-b5c0-38b4be614356

From finder

https://github.com/raycast/extensions/assets/38493346/bc59c44e-8a32-45fd-bf1c-1c8cec42b9af

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/crossbell/src/upload-file.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import { Clipboard, showToast, getSelectedFinderItems, showHUD, closeMainWindow } from "@raycast/api";
+import fs from "fs";
+import fsp from "fs/promises";
+import { ipfsUploadFile } from "crossbell/ipfs";
+import { fetch, FormData } from "node-fetch-native";
```
Comment (Songkeys) [inline extensions/crossbell/src/upload-file.tsx:5]: I don't think we need this cuz node 18+ has `fetch` built-in.

File: extensions/crossbell/src/upload-file.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      const filePaths = selectedItems.map((item) => item.path).filter((path) => fs.lstatSync(path).isFile());
+      if (filePaths.length > 0) {
+        const files = await Promise.all(
+          filePaths.map(async (path) => {
+            const data = await fsp.readFile(path);
+            return { name: path, data };
+          }),
+        );
+        console.log({ files });
+        const results = await Promise.all(
```
Comment (Songkeys) [inline extensions/crossbell/src/upload-file.tsx:26]: 1. I'm worried that there will be a huge load of files. Would a request queue be better? For example, use `p-limit`.
2. Should we use `Promise.allSettled` to handle failed requests?

**Review by Songkeys — CHANGES_REQUESTED** [review]

File: extensions/crossbell/src/upload-file.tsx (hunk 1 above)
Comment (hyoban) [reply extensions/crossbell/src/upload-file.tsx:5]: Tested on my local machine, there is no fetch and FormData, so I added node-fetch-native.

File: extensions/crossbell/src/upload-file.tsx (hunk 2 above)
Comment (hyoban) [reply extensions/crossbell/src/upload-file.tsx:26]: Oh yes, I will make a change here

**Review by Songkeys — APPROVED** [review]

**Review by mil3na — CHANGES_REQUESTED** [review]
Hi!
Thank you for your contribution! My name is Milena and I will be reviewing it today.

A few comments:
- The upload file command doesn't specify where the file will be coming from. By activating it, I only get a "No file found to upload" error. Oddly enough, I also get a green status.
- It would be interesting to either split the command in two (upload from the clipboard and from finder) or add a step. Right now it is a bit confusing to the user.
- Please, add this new command to the README

**Review by mil3na — APPROVED** [review]
LGTM!

**Pull request merged**